File: komongo2/frontend/src/domain/item/Item.ts
export interface Item {
  id: string | null
  name: string
  description: string
  createdAt: string
  updatedAt: string
}

export type ItemCreate = Omit<Item, 'id' | 'createdAt' | 'updatedAt'> & {
  id?: null
  createdAt?: string
  updatedAt?: string
}

export type ItemUpdate = Partial<Pick<Item, 'name' | 'description'>>

File: komongo2/backend/src/main/kotlin/com/sleekydz86/komongo2/item/domain/Item.kt
package com.sleekydz86.komongo2.item.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document(collection = "items")
data class Item(
    @Id
    val id: String? = null,
    val name: String,
    val description: String = "",
    @Indexed
    val createdAt: Instant = Instant.now(),
    val updatedAt: Instant = Instant.now()
)
